controller/volume_controller.go

```diff
@@ -198,16 +198,20 @@ func (vc *VolumeController) handleErr(err error, key interface{}) {
 	}
 
 	if vc.queue.NumRequeues(key) < maxRetries {
-		logrus.Warnf("Error syncing Longhorn volume %v: %v", key, err)
+		vc.logger.WithError(err).Warnf("Error syncing Longhorn volume %v", key)
 		vc.queue.AddRateLimited(key)
 		return
 	}
 
 	utilruntime.HandleError(err)
-	logrus.Warnf("Dropping Longhorn volume %v out of the queue: %v", key, err)
+	vc.logger.Warnf("Dropping Longhorn volume %v out of the queue: %v", key, err)
 	vc.queue.Forget(key)
 }
 
+func getLoggerForVolume(logger logrus.FieldLogger, v *longhorn.Volume) logrus.FieldLogger {
+	return logger.WithField("volume", v.Name).WithField("frontend", v.Spec.Frontend)
```

Comment:
this is good, add the following fields:
- withField("state", v.Status.State)
- withField("owner", v.Status.OwnerID)